File: python/fundamentals/day-01/problem3.py
# Program to convert seconds into minutes and seconds
# using floor division and remainder

def convert_seconds(total_seconds):
    # Validate input type and value
    if not isinstance(total_seconds, int) or total_seconds < 0:
        raise ValueError("Please enter a non-negative integer for seconds.")
    
    # divmod returns (quotient, remainder) in one step
    minutes, seconds = divmod(total_seconds, 60)
    return minutes, seconds

if __name__ == "__main__":
    try:
        # Input from user
        total_seconds = int(input("Enter total seconds: "))
        
        minutes, seconds = convert_seconds(total_seconds)
        print(f"{total_seconds} seconds = {minutes} minute(s) and {seconds} second(s)")
    
    except ValueError as e:
        print(f"Error: {e}")

File: java/fundamentals/day01/problem3.java
package fundamentals.day01;
import java.util.Scanner;

public class problem3 {
	public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        try {
            // Input seconds
            System.out.print("Enter total seconds: ");
            if (!sc.hasNextInt()) {
                System.out.println("Invalid input. Please enter an integer value.");
                return;
            }
            int totalSeconds = sc.nextInt();

            // Handle negative input gracefully
            if (totalSeconds < 0) {
                System.out.println("Seconds cannot be negative.");
                return;
            }

            // Floor division to get minutes
            int minutes = Math.floorDiv(totalSeconds, 60);

            // Remainder to get leftover seconds
            int seconds = totalSeconds % 60;

            // Output result
            System.out.printf("%d seconds = %d minute(s) and %d second(s)%n",
                    totalSeconds, minutes, seconds);

        } catch (Exception e) {
            System.out.println("An error occurred: " + e.getMessage());
        } finally {
            sc.close();
        }
    }
}

